WebDriverAgentLib/Categories/XCUIElement+FBUtilities.m

```diff
@@ -39,14 +39,16 @@ - (BOOL)fb_waitUntilFrameIsStable
   __block CGRect frame = self.frame;
   // Initial wait
   [[NSRunLoop currentRunLoop] runUntilDate:[NSDate dateWithTimeIntervalSinceNow:0.1]];
-  return
+  BOOL result =
   [[[FBRunLoopSpinner new]
     timeout:10.]
    spinUntilTrue:^BOOL{
-    const BOOL isSameFrame = FBRectFuzzyEqualToRect(self.frame, frame, FBDefaultFrameFuzzyThreshold);
-    frame = self.frame;
+    CGRect newFrame = self.frame;
+    const BOOL isSameFrame = FBRectFuzzyEqualToRect(newFrame, frame, FBDefaultFrameFuzzyThreshold);
+    frame = newFrame;
     return isSameFrame;
   }];
+  return result;
```

Comment:
is this change necessary?

Reply:
Calling `self.frame` retrieves a snapshot of the element every time.

From my testing saving the result of the frame instead of calling `self.frame` every time the property is needed is slightly faster (5% faster in my simple stress scenario); I've experimented with ways to make the possible performance impact lower, also tested using `fb_snapshotwithattributes` with only the frame as a parameter, but it didn't make much of a difference from just calling `self.frame`.

Saving the result into a variable was just for printing execution times though, I'll remove it.